### PR TITLE
Fix test so it doesn't leak frontend_server process

### PR DESCRIPTION
**Pull request by aam**
flutter_run_test doesn't stop FlutterTestDriver, which results in frontend_server processes staying running in memory

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]

File: packages/flutter_tools/test/integration/flutter_run_test.dart [hunk 1]
```diff
@@ -53,6 +53,7 @@ void main() {
       final File pidFile = tempDir.childFile('test.pid');
       await _flutter.run(pidFile: pidFile);
       expect(pidFile.existsSync(), isTrue);
+      await _flutter?.stop();
```
Comment (DanTup) [inline packages/flutter_tools/test/integration/flutter_run_test.dart]: Good spot!

Do we need to ensure this runs even if the line above throws though?

Comment (aam) [reply packages/flutter_tools/test/integration/flutter_run_test.dart]: If there is no pidFile, is there a process running? `_flutter.stop()` doesn't like being called when process is not running(like in test above).

Comment (DanTup) [reply packages/flutter_tools/test/integration/flutter_run_test.dart]: If the process runs but doesn't create the pid-file (which is the bug that resulted in this test), the process would've still been created so the `stop()` method should be good to call (it skips most of its code if the debugger never attached/app never started and just waits on `_proc.exitCode`).

Comment (aam) [reply packages/flutter_tools/test/integration/flutter_run_test.dart]: Okay, got you.

**Pull request merged**